test/verify/testsuite-prepare

```diff
@@ -1,87 +1,69 @@
 
-usage()
-{
-    echo "usage: testsuite-prepare [--clean] [--quick] [--verbose]"
-}
+import argparse
+import shutil
+import os
+import glob
+import shutil
 
+import parent
+from common import testinfra
+from common import vmcommon
 
-args=$(getopt -o "h,v,q" -l "help,verbose,quick" -- "$@")
-eval set -- "$args"
-while [ $# -gt 0 ]; do
-    case $1 in
-    -q|--quick)
-        vm_install_opts="$vm_install_opts --quick"
-        ;;
-    -v|--verbose)
-        vm_install_opts="$vm_install_opts --verbose"
-        ;;
-    -h|--help)
-        usage
-        exit 0
-        ;;
-    --)
-        shift
-        break
-        ;;
-    esac
-    shift
-done
+image_list=["ipa", "openshift", "fedora-stock"]
 
-# The Atomic variants can't build their own packages, so we build in
-# their non-Atomic siblings.  For example, fedora-atomic is build
-# in fedora-23
-#
-if [ "$TEST_OS" = "fedora-atomic" ]; then
-    BUILD_OS="fedora-23"
-elif [ "$TEST_OS" = "rhel-atomic" ]; then
-    BUILD_OS="rhel-7"
-else
-    BUILD_OS="$TEST_OS"
-fi
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+            description='Prepare testig environment, download images and build and cinstall cockpit',
+            formatter_class=argparse.ArgumentDefaultsHelpFormatter
+        )
+    parser.add_argument('-v', '--verbose', action='store_true', help='Display verbose progress details')
+    parser.add_argument('-n', '--noinstall', action='store_true', help='Do not install cockpit to base image')
+    parser.add_argument('-f', '--force', action='store_true', help='Force update of images')
+    parser.add_argument('image', nargs='?', default=testinfra.DEFAULT_IMAGE, help='The image to use')
+    args = parser.parse_args()
 
-$BASE/vm-reset
-$BASE/vm-download \
-    $TEST_OS \
-    $BUILD_OS \
-    ipa \
-    openshift \
-    fedora-stock
+    test_os = args.image
+    build_os = test_os
 
-install_failed=no
-if ! $BASE/vm-install $vm_install_opts --build-image $BUILD_OS; then
-    install_failed=yes
-fi
+    # The Atomic variants can't build their own packages, so we build in
+    # their non-Atomic siblings.  For example, fedora-atomic is build
+    # in fedora-23
+    if test_os == "fedora-atomic":
+        build_os = "fedora-23"
+    elif test_os == "rhel-atomic":
+        build_os = "rhel-7"
 
-# Upload the build logs if desired
-if [ -n "${TEST_ATTACHMENTS:-}" ]; then
-    mkdir "$TEST_ATTACHMENTS"/build-results
-    cp -r $BASE/tmp/build-results/*.log "$TEST_ATTACHMENTS"/build-results
-fi
+    os.system(os.path.join(testinfra.TEST_DIR,"vm-reset"))
 
-if [ $install_failed == "yes" ]; then
-  exit 1
-fi
+    vmcommon.download_images(set((test_os, build_os)), args.force, [])
+    vmcommon.download_images(image_list, args.force, [])
 
-exit 0
+    if args.noinstall:
+        print "Not installing cockpit to base image"
+    else:
+        vmcommon.build_and_install(test_os, build_os, {"verbose":args.verbose, "force": args.force})
```

Comment:
When the build fails, this will throw an exception and terminate `testsuite-preprare`.  The exception needs to be caught so that we attach the build logs below.  Also, `testsuite-prepare` needs to exit non-zero when the build failed.

I think it is best to make `vmcommon.build_and_install` return `False` when the build fails.